api/scpca_portal/models/job.py

```diff
@@ -104,6 +121,59 @@ def get_sample_job(
             },
         )
 
+    @classmethod
+    def bulk_sync_state(cls) -> bool:
+        """
+        Sync all submitted job instances' states with AWS Batch job statuses.
+        Call batch.get_jobs to fetch all the corresponding remote jobs.
+        Update each job instance's state if it changes to COMPLETED, and update completed_at.
+        If the remote status is 'FAILED', update failure_reason if it hasn't been set already.
+        """
+        if submitted_jobs := cls.objects.filter(state=JobStates.SUBMITTED):
+            batch_job_ids = [job.batch_job_id for job in submitted_jobs]
+
+            if fetched_jobs := batch.get_jobs(batch_job_ids):
+                # Map the fetched AWS jobs for easy lookup by batch_job_id
+                aws_jobs = {job["jobId"]: job for job in fetched_jobs}
+
+                synced_jobs = []
+
+                for job in submitted_jobs:
+                    if aws_job := aws_jobs.get(job.batch_job_id):
+                        aws_job_status = aws_job["status"]
+                        is_terminated = aws_job.get("isCancelled") or aws_job.get("isTerminated")
+                        mapped_job_state = job.get_mapped_job_state(
+                            aws_job_status, is_terminated=is_terminated
+                        )
+
+                        # Only update the job if the state has changed
+                        if job.state != mapped_job_state:
+                            if aws_job_status == "FAILED" and not (
+                                is_terminated or job.failure_reason
+                            ):
+                                job.failure_reason = aws_job["statusReason"]
+
+                            if is_terminated:
+                                job.terminated_at = make_aware(datetime.now())
+
+                            job.state = mapped_job_state
+                            job.completed_at = make_aware(datetime.now())
+                            synced_jobs.append(job)
+                    else:
+                        # TODO: How should we handle when no matched AWS job returned?
+                        # Currently, it continues without interruption - log, or raise an error?
+                        continue
```

Comment:
While implementing the tests, I encountered an error due to a missing matched batch job (thanks to my typo 😅). This made me consider the scenario where no matching AWS job is found via the API call. Although this may be unlikely, if it does happen, it could halt the execution of the bulk syncing process. Just to be safe, should we handle this case?